### ISSUE TITLE
[ecmascript] function declaration report `unnecessary block` issue

### ISSUE THREAD

**Issue by naveadkazi** [pmd/pmd#4129]
**Affects PMD Version:** Not sure, in codacy it give since 5.0

**Rule:**

Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/latest/pmd_rules_ecmascript_codestyle.html#unnecessaryblock

**Description:**
Created a named exported function that returns an object. This function is pushed to codacy.com which uses PMD. Codacy reports it as an unnecessary block. I am not sure if this false positive and what I am missing here.

![image](https://user-images.githubusercontent.com/12540014/190968165-a5150742-401b-4dcd-9d1a-b28da5c498f5.png)

**Code Sample demonstrating the issue:**

```

```

**Expected outcome:**

PMD reports a violation at line 188 in the screenshot, but that's wrong. That's a false positive.

**Running PMD through:** Codacy

**Comment by naveadkazi** [comment]
EDIT: codacy using PMD :  6.48.0

**Comment by naveadkazi** [comment]
I tried running on local environment and it reported as `Unnecessary block`. But if I move the `export` statement at end of the file then PMD is not reporting it. The below code is similar to one in the issue description with named export at the end. This is not reporting the issue

```
const getAddtitonalTagTrackingImpressionObj = (
  profileId,
  cityName,
  tracking,
  isAdditionalTagsAndSlicerAbTest
) => {
  const action = isAdditionalTagsAndSlicerAbTest
    ? `TagShown${tracking}`
    : ADDITIONAL_TAG_NOT_SHOWN_ACTION;
  return {
    gaCat: SEARCH_PAGE_TRACKING_CATEGORY,
    gaActImp: action,
    gaLblImp: `profileId=${profileId}|Cityname=${cityName}`,
    once: true,
    trackImpression: true,
  };
};

export { getAddtitonalTagTrackingImpressionObj };
```

**Comment by oowekyala** [comment]
Related to #2305. Probably `export` is not supported by Rhino, or at least improperly parsed. The false-positive could probably be fixed by also requiring that the reported block is not a child of the root node (see the rule's xpath definition linked above).
